### ISSUE TITLE
DOC Identify and list components of the API that could benefit from short examples in the documentation

### ISSUE THREAD

**Issue by TamaraAtanasoska** [fairlearn/fairlearn#1475]
#### Describe the issue linked to the documentation
The [Fairlearn API](https://fairlearn.org/v0.12/api_reference/index.html) still has many of it's components not covered by short examples that show their functionality in the [User Guide](https://fairlearn.org/v0.12/user_guide/index.html). 

#### Suggest a potential alternative/fix
Adding more examples of example usage of the different components of the API in the User Guide.
Some of the examples can be added within the existing documents, but the User Guide can also be expanded. 

**A big step forward would be go to through the API reference and identify which components aren't already in the User Guide and create a list as part of this issue.** 

While not directly merging anything, this issue provides great value for the project. The benefit for the contributor is getting familiar with the API in detail and easily picking up subsequent tasks. 

An example of such a short example in the user guide of a PR we merged recently: 
- [Place in the docs where it resides](https://fairlearn.org/v0.12/user_guide/mitigation/reductions.html#exponentiated-gradient)
- The PR discussion around it: https://github.com/fairlearn/fairlearn/pull/1353

Feel free to ping me with any questions! You can also go ahead and open a PR about adding an example that is missing.

**Comment by StefanieSenger** [comment]
I am working on adding an example for `ErrorRate`.

**Comment by TamaraAtanasoska** [comment]
Hi @StefanieSenger, are there maybe some functions/classes you came upon searching what to do that you decided not to go into and you could list here? It is ok if not, maybe we can leave a breadcrumb for the next person. Thanks!

**Comment by alliesaizan** [comment]
Hi @TamaraAtanasoska, this issue is pretty similar to #987 opened by @hildeweerts. Would it make sense to consolidate these two issues?

**Comment by TamaraAtanasoska** [comment]
yes they definitely can be consolidated into a larger project. both the tasks are still relevant though, as a list with all the missing information make it more tangible for the newcomers to the project.

**Comment by alliesaizan** [comment]
> yes they definitely can be consolidated into a larger project. both the tasks are still relevant though, as a list with all the missing information make it more tangible for the newcomers to the project.

Understood

**Comment by 3Paradox** [comment]
Hi, I noticed make_derived_metric doesn't have an example in the User Guide. Ran into this while building a churn model. Adding one with transform='difference' might be useful. Happy to work on this if no one else is on it!

**Comment by 3Paradox** [comment]
Actually, I looked more carefully and saw it already has an example in custom_fairness_metrics.rst. I'll check the other components and follow up with something that's actually missing

**Comment by 3Paradox** [comment]
Opened PR #1637 for mean_prediction

**Comment by 3Paradox** [comment]
It looks like the CI failures are due to plot_credit_loan_decisions.py hitting an HTTP 502 when fetching external data, which seems unrelated to this PR.
The advanced_metricframe doctest passes (50/50), and the added example builds correctly.
